### ISSUE TITLE
Consider using rls-span

### ISSUE THREAD

**Issue by nrc** [phildawes/racer#661]
https://github.com/nrc/rls-span

This crate offers an array of types for defining positions and ranges in source text. Using it has two advantages for Racer: statically protects against off-by-one errors by using phantom types; easier interoperation with the RLS.

I'm not entirely sure the payoff is worth the effort of moving, but I have found it pleasant to be explicit about zero- or one-based offsets.

**Comment by jwilm** [comment]
I like the explicitness about 0/1-based indexing. Not sure the work required to make racer use this everywhere makes sense; in addition to being work for racer, it would be work for all the dependents. Instead, what do you think about some adapters to wrap racer APIs?

**Comment by nrc** [comment]
Yeah, it is quite a lot of work. We basically wrap the Racer types in these types in the client (RLS), I'm not sure if it is better engineering to wrap there or in Racer. I don't think it makes a big difference either way.
